docs/adrs/0004-field-for-institution-information.md

```diff
@@ -0,0 +1,68 @@
+
+Follows: https://mitlibraries.github.io/guides/misc/adr.html
+
+Date: 2024-02-22
+
+## Status
+
+Proposed.
+
+## Context
+
+The UI for GeoData needs to display the host institution of non-MIT records (e.g., "Stanford", "University of Wisconsin-Madison") to signal to end users where an outlink is going. The question was raised of _where_ this information can be found in the existing TIMDEX data model or, if not present, what field will be used to store this information.
+
+### Current mappings
+
+| Source | Mappings | TIMDEX Field | Examples       |
+| ------ | -------- | ------------ | -------------- |
+| Alma   | [MARC$985$i](https://www.oclc.org/bibformats/en/9xx.html) | Holdings.Location | Hayden Library |
+| MIT GIS | [schema_provider_s](https://opengeometadata.org/ogm-aardvark/#provider)| PublicationInformation | GIS Lab, MIT Libraries |
+| OGM GIS | [schema_provider_s](https://opengeometadata.org/ogm-aardvark/#provider)| PublicationInformation | University of Wisconsin-Milwaukee |
+
+1. **Does `geo-harvester` collect the required information (i.e., host institution)?**
+
+   **Yes.** The Aardvark metadata schema includes a field called `schema_provider_s`, which is a single string value describing the [“organization [that] holds the resource or acts as the custodian for the metadata record and to help users understand which resources they can access”](https://opengeometadata.org/ogm-aardvark/#provider). 
+
+   The derivation in `geo-harvester` sets `schema_provider_s` to "GIS Lab, MIT Libraries" for MIT records and the assigned name for the repo in the [OGM repositories config YAML file](https://github.com/MITLibraries/geo-harvester/blob/main/harvester/ogm_repositories_config.yaml).
+
+2. **Does the `MITAardvark` transformer map the required information to a field in the TIMDEX data model?**
+
+   **Yes, but its current mapping does not make the value easily accessible.** In `transmogrifier`, [the value from `schema_provider_s` is appended to `TimdexRecord.publication_information`](https://github.com/MITLibraries/transmogrifier/blob/main/transmogrifier/sources/json/aardvark.py#L382-L383), which comprises of a list of strings describing the publishing details for a given record. The contents of this field will vary by source and depending on what data is available. Trying to derive which value--from a list of strings (without any sort of identifier)--came from `schema_provider_s` is a nontrivial task.
+
+## Decision
+
+**A new field called "institution" is added to the TIMDEX data model** (i.e., `TimdexRecord.institution`). This field will denote--as its name implies--the institution or organization that provides access to the resource described in the TIMDEX record. This field exists at the top level of the TIMDEX data model, making it easily accessible for referencing in the UI or querying. This solution also avoids using existing fields (discussed below) in ways that obscure the TIMDEX data model.
```

Comment:
On a closer reread, I do think this does a good  job at proposing what this field "means":

> "This field will denote--as its name implies--the institution or organization that provides access to the resource described in the TIMDEX record"

But, if we are thinking of the reusability or naming of this new field, perhaps that "access" word is important.  Is it possible this is really about access primarily?  If so, what if the field were something along the lines of `access_provider`?  Or even just `provider`?

Reply:
I think `provider` is an interesting concept to consider here. For GIS it seems to map clearly because the provider of the metadata and provider of the resource are generally the same from the TIMDEX perspective.

It seems to get messier for subscription resources that might appear in Alma (which might either help us refine the intent of this field... even if our finding is that this field is not to be used for vended resources :) ). Would the `provider` be the vendor that supplies the content or the subscriber to the content on behalf of the user. i.e. MIT subscribes to an Ebsco database so our users can access the content. Is MIT the `provider` or is `ebsco` the provider. As a consumer service, TIMDEX API should likely take the user perspective?

Reply:
As a consumer service, and a _discovery_ one at that, I suppose I'd expect `provider` to indicate what instituion/organization/company is providing physical or digital access to the resource.

> ...i.e. MIT subscribes to an Ebsco database so our users can access the content. Is MIT the provider or is ebsco the provider.

I like this example because I think a) `provider="Ebsco"` would be ideal, but b) I think that level of provider granularity may vary considerably from source-to-source.  Examples:

- Libguides / Research Database
  - I've confirmed that OAI-PMH records offer no indication of the "leaf" node of access
  - Research Databases: might consider a default of `provider="MIT"`
  - LibGuides: we could say `provider="Springshare"`
- DSpace
  - feels like we could default to `provider="MIT"` for all records
- Alma
  - probably the most complex, where records may have varying degrees of granularity here
- ArchivesSpace
  - perhaps another good example where `provider="MIT"` default makes sense

Don't want to dig too far into the logic, as `provider` is just a consideration at the moment.  But unless we explicitly know -- from a consumer / discovery POV -- maybe a default of `provider="MIT"` wouldn't be inaccurate.

Reply:
I may lean more towards `provider=unknown` for things like Alma/LibGuides that we aren't sure on... and then take that unknown data to the metadata team and ask if they could make that clearer in the records or how we could better detect it/supplement records from non-alma sources, etc. The reason I'd suggest not defaulting to "MIT" is that would make it very difficult to understand which are actually MIT and which are :shrug:

Reply:
Good point, makes sense.  And maybe to get a bit technical, we could do `provider=NULL`?  Understanding that the absence of a field value means we don't know for certain?  

And clarification: when you say "LibGuides" above, do you mean "ResearchDatabases" (AZ list)?  As for "LibGuides" (they are distinct sources in TIMDEX), it feels like we _do_ know that the provider is always Springshare.

Reply:
Oh sorry, I read misread and thus misrepresented. LibGuides is probably either MITL or Springshare. The content is curated/created by MITL but using the Springshare platform.

Research Databases is what I meant to be replying to with the "unknown" or your follow on idea of just not setting it (i.e. null).